### DISCUSSION TITLE
Is it possible to make defaultHeaders mutable?

### DISCUSSION THREAD

**Discussion by beva-sdev** [AnWeber/vscode-httpyac#199] [Q&A]
Dear all,
the API I need to test has in 90% cases headers in form:
 - Authorization: Bearer {{token}}
 - Content-Type: application/json

Just the get Token (the first request) and revoke token (the lastrequest) are using:

 - Authorization: Basic user:pw
 - Content-Type: application/x-www-form-urlencoded

Did following:
1) set the defaultHeaders in VS-Code setting.json for the 90% of requests 
2) for request with another another requirements I have specified the header in the request Block. 

Problem: the per request headers doesnt overwrite the default ones.
It seems the defaultHeaders are immutable.

Do we have any solution/workaround?

Thanks in advance

**Comment by AnWeber** [comment]
@beva-sdev  Behavior can be produced with different case of the headers. These must be taken into account when overwriting. Can it be because of this. Otherwise, I would actually already note this, I would have claimed.

**Reply by beva-sdev** [reply]
Thanks,  indeed, since yesterday works well as expected. No idea what I did wrong, it was apparently not my day.

Thanks
